internal/tools/pipeline/providers/leaderworker/impl_general.go

```diff
@@ -30,12 +30,23 @@ func (p *provider) ListWorkers(ctx context.Context, workerTypes ...worker.Type)
 }
 
 func (p *provider) ListenPrefix(ctx context.Context, prefix string, putHandler, deleteHandler func(context.Context, *clientv3.Event)) {
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
add a info log to tip stop listen prefix

Reply:
done